### PR TITLE
feat: add an 'optional' field to ConnectionAnnotation

### PR DESCRIPTION
**Pull request by gerfboy**

### CODE REVIEW

File: gateway/src/main/java/com/mx/path/gateway/configuration/annotations/Connection.java [hunk 1]
```diff
@@ -13,6 +13,13 @@
...
 
+  /**
+   * The connection name to be bound from gateway configuration.
+   */
   String value();
 
+  /**
+   * Indicates whether the connection is optionally required to be configured in the gateway configuration. Default is false.
+   */
+  boolean optional() default false;
```
Comment (mattnichols) [inline gateway/src/main/java/com/mx/path/gateway/configuration/annotations/Connection.java:24]: I think we use `required` in other places. I think that would be clearer. Sorry, I think the rough idea came out with an optional attribute. 😆

Comment (mattnichols) [reply gateway/src/main/java/com/mx/path/gateway/configuration/annotations/Connection.java:24]: Default to `true`

Comment (gerfboy) [reply gateway/src/main/java/com/mx/path/gateway/configuration/annotations/Connection.java:24]: After some discussion we think 'optional' works as well.

Comment (gerfboy) [reply gateway/src/main/java/com/mx/path/gateway/configuration/annotations/Connection.java:24]: Just to be clear: I'm sticking with @Connection(value = "name", optional = true) to declare a connection may come in null.

@Connection("name") will throw an exception if the connection is not configured.

This means that an accessor that doesn't specify a connection is optional won't have to null-check the connection in the code.

Comment (mattnichols) [reply gateway/src/main/java/com/mx/path/gateway/configuration/annotations/Connection.java:24]: Agreed.

**Review by mattnichols — APPROVED** [review]

**Pull request merged**